### PR TITLE
Support record data with DeleteRecordSet

### PR DESCRIPTION
**Pull request by mitruly**
Support multi-record by allowing optional `record` data field for `DeleteRecordSet`.

Changes in this PR:
- Add `record` field to `ChangeInput`, refactoring `DeleteRecordSetChangeInput` and `AddChangeInput`
- Ensure `record` field in `DeleteRecordSet` data models
- Include `DeleteRecordSetSingleChange` in GSON adapter factory
- Update unit test

------
Tested against local instance of API:
```
List<ChangeInput> changes = new ArrayList<>();
    changes.add(new DeleteRecordSetChangeInput("doot-java.ok", RecordType.A));
    changes.add(new DeleteRecordSetChangeInput("doot-java-with-data.ok.", RecordType.A, new AData("1.2.3.4")));

CreateBatchRequest req =
    new CreateBatchRequest(
        "vinyldns-java test with and without record data",
        changes,
        "duGroup-id");
VinylDNSResponse<BatchResponse> res = testUserClient.createBatchChanges(req);
System.out.println("response: " + res);
```

Response:
```
response: VinylDNSSuccessResponse{value=BatchResponse{id='a579a728-d7a1-4ffc-af93-c81fb28c3f92', userId='testuser', userName='testuser', comments='vinyldns-java test with no record data', createdTimestamp=Thu Oct 10 15:37:57 EDT 2019, changes=[DeleteRecordSetSingleChange{id='a11308b3-3868-44fe-8acc-5c88c52912f0', changeType=DeleteRecordSet, zoneId='', zoneName='', recordName='', inputName='doot-java.ok.', type=A, status=NeedsReview, systemMessage='null', recordChangeId='null', recordSetId='null', record=null, validationErrors=[ValidationError{errorType=ZoneDiscoveryError, message='Zone Discovery Failed: zone for "doot-java.ok." does not exist in VinylDNS. If zone exists, then it must be connected to in VinylDNS.'}]}, DeleteRecordSetSingleChange{id='8d9b01f5-82cb-4fbc-ab8e-b8fbe0dca47e', changeType=DeleteRecordSet, zoneId='', zoneName='', recordName='', inputName='doot-java-with-data.ok.', type=A, status=NeedsReview, systemMessage='null', recordChangeId='null', recordSetId='null', record=AData{address='1.2.3.4'}, validationErrors=[ValidationError{errorType=ZoneDiscoveryError, message='Zone Discovery Failed: zone for "doot-java-with-data.ok." does not exist in VinylDNS. If zone exists, then it must be connected to in VinylDNS.'}]}], status=PendingReview, ownerGroupId='duGroup-id', ownerGroupName='null', approvalStatus=PendingReview, reviewerId='null', reviewComment='null', reviewTimestamp=null, scheduledTime=null, cancelledTimestamp=null}, messageBody={"userId":"testuser","userName":"testuser","comments":"vinyldns-java test with no record data","createdTimestamp":"2019-10-10T19:37:57Z","changes":[{"changeType":"DeleteRecordSet","inputName":"doot-java.ok.","type":"A","status":"NeedsReview","recordName":"","zoneName":"","zoneId":"","validationErrors":[{"errorType":"ZoneDiscoveryError","message":"Zone Discovery Failed: zone for \"doot-java.ok.\" does not exist in VinylDNS. If zone exists, then it must be connected to in VinylDNS."}],"id":"a11308b3-3868-44fe-8acc-5c88c52912f0"},{"changeType":"DeleteRecordSet","inputName":"doot-java-with-data.ok.","type":"A","record":{"address":"1.2.3.4"},"status":"NeedsReview","recordName":"","zoneName":"","zoneId":"","validationErrors":[{"errorType":"ZoneDiscoveryError","message":"Zone Discovery Failed: zone for \"doot-java-with-data.ok.\" does not exist in VinylDNS. If zone exists, then it must be connected to in VinylDNS."}],"id":"8d9b01f5-82cb-4fbc-ab8e-b8fbe0dca47e"}],"status":"PendingReview","approvalStatus":"PendingReview","id":"a579a728-d7a1-4ffc-af93-c81fb28c3f92","ownerGroupId":"duGroup-id"}, statusCode=202}
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>io.vinyldns</groupId>
     <artifactId>vinyldns-java</artifactId>
-    <version>0.9.3-SNAPSHOT</version>
+    <version>0.9.4-SNAPSHOT</version>
```
Comment (mitruly) [inline pom.xml]: Seems like we didn't update this last version release.

File: src/main/java/io/vinyldns/java/model/batch/AddChangeInput.java [hunk 2]
```diff
@@ -19,27 +19,21 @@
 
 public class AddChangeInput extends ChangeInput {
   private final Long ttl;
-  private final RecordData record;
```
Comment (mitruly) [inline src/main/java/io/vinyldns/java/model/batch/AddChangeInput.java]: Consolidating shared `RecordData record` field into `ChangeInput` class.

File: src/main/java/io/vinyldns/java/model/batch/ChangeInput.java [hunk 3]
```diff
@@ -39,18 +50,21 @@ public RecordType getType() {
...
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     ChangeInput that = (ChangeInput) o;
     return this.inputName.equals(that.inputName)
         && this.changeType == that.changeType
-        && this.type == that.type;
+        && this.type == that.type
+        && this.record == that.record;
```
Comment (pauljamescleary) [inline src/main/java/io/vinyldns/java/model/batch/ChangeInput.java]: `==` in java for objects tests that they point to the same reference, you will want `this.record.equals(that.record)` here.

Also, keep in mind `this.record` and `that.record` could be null, so simply `this.record.equals(that.record)` is not safe.  Need a stronger check...

```java
boolean recordsAreEqual = ((this.record != null && this.record.equals(that.record)) || (that.record != null && that.record.equals(this.record)) || (this.record == null && that.record == null))
```

Comment (pauljamescleary) [reply src/main/java/io/vinyldns/java/model/batch/ChangeInput.java]: Make sure once you update that you add some good unit tests around that

Comment (pauljamescleary) [reply src/main/java/io/vinyldns/java/model/batch/ChangeInput.java]: Hmmm, down below we see `Objects.equals`, maybe that is the right approach?

**Pull request merged**